performance/tests/Button.test.js

```diff
@@ -1,82 +1,139 @@
+			await page.waitForTimeout(100);
 			await page.mouse.up();
 			await page.mouse.down();
-			await page.waitFor(200);
+			await page.waitForTimeout(100);
 			await page.mouse.up();
 			await page.mouse.down();
-			await page.waitFor(200);
+			await page.waitForTimeout(100);
 			await page.mouse.up();
 			await page.mouse.down();
-			await page.waitFor(200);
+			await page.waitForTimeout(100);
 			await page.mouse.up();
 
-			await page.tracing.stop();
-
-			const actualFPS = FPS(filename);
-			TestResults.addResult({component: 'Button', type: 'Frames Per Second', actualValue: actualFPS});
-
-			const actualUpdateTime = Update(filename, 'Touchable');
-			TestResults.addResult({component: 'Button', type: 'Update', actualValue: actualUpdateTime});
+			const averageFPS = (FPSValues.reduce((a, b) => a + b, 0) / FPSValues.length) || 0;
+			TestResults.addResult({component: component, type: 'Frames Per Second Click', actualValue: averageFPS});
 		});
 	});
 
 	describe('keypress', () => {
 		it('animates', async () => {
-			const filename = getFileName('Button');
-
+			const FPSValues = await FPS();
 			await page.goto('http://localhost:8080/button');
-			await page.tracing.start({path: filename, screenshots: false});
 			await page.waitForSelector('#button');
 			await page.focus('#button');
-			await page.waitFor(200);
+			await page.waitForTimeout(100);
 			await page.keyboard.down('Enter');
-			await page.waitFor(200);
+			await page.waitForTimeout(100);
 			await page.keyboard.up('Enter');
 			await page.keyboard.down('Enter');
-			await page.waitFor(200);
+			await page.waitForTimeout(100);
 			await page.keyboard.up('Enter');
 			await page.keyboard.down('Enter');
-			await page.waitFor(200);
+			await page.waitForTimeout(100);
 			await page.keyboard.up('Enter');
 			await page.keyboard.down('Enter');
-			await page.waitFor(200);
+			await page.waitForTimeout(100);
 			await page.keyboard.up('Enter');
 
-			await page.tracing.stop();
+			const averageFPS = (FPSValues.reduce((a, b) => a + b, 0) / FPSValues.length) || 0;
+			TestResults.addResult({component: component, type: 'Frames Per Second Keypress', actualValue: averageFPS});
+		});
+	});
+
+	it('should have a good FID and CLS', async () => {
+		await page.evaluateOnNewDocument(FID);
+		await page.evaluateOnNewDocument(CLS);
+		await page.goto('http://localhost:8080/button');
+		await page.waitForSelector('#button');
+		await page.focus('#button');
+		await page.keyboard.down('Enter');
+		await page.waitForTimeout(200);
 
-			const actualFPS = FPS(filename);
-			TestResults.addResult({component: 'Button', type: 'Frames Per Second', actualValue: actualFPS});
+		let actualFirstInput = await page.evaluate(() => {
+			return window.fid;
+		});
 
-			const actualUpdateTime = Update(filename, 'Touchable');
-			TestResults.addResult({component: 'Button', type: 'Update', actualValue: actualUpdateTime});
+		let actualCLS = await page.evaluate(() => {
+			return window.cls;
```

Comment:
Can't we reuse this function in one place rather than duplicate it in each component's code?